media.go

```diff
@@ -239,6 +255,8 @@ func (pc *producerConsumer[T, U]) start() {
 				return
 			}
 
+			var doReadSpan *trace.Span
+			startLocalCtx, doReadSpan = trace.StartSpan(startLocalCtx, "gostream::producerConsumer (anonymous function to read)")
```

Comment:
Why not do this in the anonymous function below and add `doReadSpan.End()` to the defer func in that same function after line 265?

Reply:
Also definitely an oversight on my part!